flashcards/views.py

```diff
@@ -1,16 +1,205 @@
+    template_name = 'confirm_card_deletion.html'
+
+    @property
+    def deck_id(self):
+        return self.kwargs['pk']
+
+    @property
+    def card_id(self):
+        return self.kwargs['card_pk']
+
+    def get_object(self, queryset=None):
+        return Flashcard.objects.get(id=self.card_id)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['pk'] = self.deck_id
+        context['card_pk'] = self.card_id
+        return context
+
+    def get_success_url(self):
+        return reverse_lazy('deck-detail', kwargs={'pk': self.deck_id})
+
+
+class CardUpdateView(LoginRequiredMixin, UpdateView):
+    fields = ['front_text', 'back_text']
+    template_name = 'update_card.html'
+
+    @property
+    def deck_id(self):
+        return self.kwargs['pk']
+
+    @property
+    def card_id(self):
+        return self.kwargs['card_pk']
+
+    def get_object(self, queryset=None):
+        return Flashcard.objects.get(id=self.card_id)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['pk'] = self.deck_id
+        context['card_pk'] = self.card_id
+        return context
+
+    def get_success_url(self):
+        return reverse_lazy('deck-detail', kwargs={'pk': self.deck_id})
 
 
 class UserDecksView(LoginRequiredMixin, TemplateView):
     template_name = 'user_decks.html'
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['decks'] = Deck.objects.all()
+        return context
+
+
+class CardCreateView(LoginRequiredMixin, CreateView):
+    model = Flashcard
+    fields = ['front_text', 'back_text']
+    template_name = 'create_card.html'
+
+    @property
+    def deck_id(self):
+        return self.kwargs['pk']
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['pk'] = self.deck_id
+        return context
+
+    def form_valid(self, form):
+        deck_form: Flashcard = form.instance
+        deck_form.last_seen_date = datetime.now()
+        deck_form.parent_deck = Deck.objects.get(pk=self.deck_id)
+
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        return reverse_lazy('deck-detail', kwargs={'pk': self.deck_id})
+
+
+class DeckDetailView(LoginRequiredMixin, TemplateView):
+    template_name = 'deck_detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        deck = Deck.objects.get(pk=context['pk'])
+        context['title'] = deck.title
+        context['cards'] = deck.flashcards.all()
+        context['deck'] = deck
+
+        return context
+
+
+class DeckCreateView(LoginRequiredMixin, CreateView):
+    model = Deck
+    fields = ['title']
+    template_name = 'create_deck.html'
+
+    def form_valid(self, form):
+        deck_form: Deck = form.instance
```

Comment:
Ooh, a type hint. Is this intentional?

Reply:
Yup, whenever I'm not 100% sure how something behaves I just add type hints and PyCharm will usually help out with auto-completions.